listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -208,94 +246,158 @@ def insert(self, listens):
             the value they are converted to the local server's timezone. So to compare
             datetime object we need to create a object in the same timezone as the server.
 
+            If neither from_ts nor to_ts is provided, the latest listens for the user are returned.
+            Returns a tuple of (listens, min_user_timestamp, max_user_timestamp)
+
             from_ts: seconds since epoch, in float
             to_ts: seconds since epoch, in float
             limit: the maximum number of items to return
             order: 0 for ASCending order, 1 for DESCending order
-            time_range: the time range (in units of 5 days) to search for listens. If none is given
-                        3 ranges (15 days) are searched. If -1 is given then all listens are searched
-                        which is slow and should be avoided if at all possible.
         """
 
-        return self.fetch_listens_for_multiple_users_from_storage([user_name], from_ts, to_ts, limit, order, time_range)
+        return self.fetch_listens_for_multiple_users_from_storage([user_name], from_ts, to_ts, limit, order)
 
-    def fetch_listens_for_multiple_users_from_storage(self, user_names: List[str], from_ts: float, to_ts: float, limit: int, order: int, time_range: int=3):
+    def fetch_listens_for_multiple_users_from_storage(self, user_names: List[str], from_ts: float, to_ts: float, limit: int, order: int):
         """ The timestamps are stored as UTC in the postgres datebase while on retrieving
             the value they are converted to the local server's timezone. So to compare
             datetime object we need to create a object in the same timezone as the server.
 
+            If neither from_ts nor to_ts is provided, the latest listens for the user are returned.
+            Returns a tuple of (listens, min_user_timestamp, max_user_timestamp)
+
             from_ts: seconds since epoch, in float
             to_ts: seconds since epoch, in float
             limit: the maximum number of items to return
             order: 0 for DESCending order, 1 for ASCending order
-            time_range: the time range (in units of 5 days) to search for listens. If none is given
-                        3 ranges (15 days) are searched. If -1 is given then all listens are searched
-                        which is slow and should be avoided if at all possible.
         """
 
-        if time_range is None:
-            time_range = 3
-
-        if time_range < 0:
-            max_timestamp_window = -1
-        else:
-            max_timestamp_window = SECONDS_IN_TIME_RANGE * time_range
-            if to_ts is None:
-                to_ts = from_ts + max_timestamp_window
-            elif from_ts is None:
-                from_ts = to_ts - max_timestamp_window
-
+        min_user_ts = max_user_ts = -1
+        for user_name in user_names:
+            min_ts, max_ts = self.get_timestamps_for_user(user_name)
```

Comment:
Is this going to come from redis the majority of the time? (I believe it is, just checking). If it's coming from the db, then this is 1 query per username, which could potentially be slow (faster to do it for all usernames in a single query).
How many usernames are typically used here? Again - if there are always many usernames it may be faster to get all keys from redis at once. But if there's normally only one then it's probably not worth optimising this lookup

Reply:
They are mostly coming from redis, yes. I see your points, but they all existed as this before this PR and I really don't want to fix this stuff now. Please make a ticket.